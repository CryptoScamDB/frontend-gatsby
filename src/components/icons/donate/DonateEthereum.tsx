import React from 'react';
import Icon from '../Icon';
import LogoEthereum from '../../../images/coins/ethereum.png';

const DONATION_ADDR = {
    "ADDR": 0x7030B43E821BA34124915bA65B90Dbda7a9E7d09,
    "HUMAN": "cryptoscamdb.eth",
    "EXPLORER": "https://etherscan.io/address/0x7030B43E821BA34124915bA65B90Dbda7a9E7d09"
}

const DonateEthereum: React.StatelessComponent = () => (
    <a href={DONATION_ADDR.EXPLORER} target="_blank" rel="noopener noreferrer">
        <Icon 
            title={DONATION_ADDR.HUMAN} 
            src={LogoEthereum} 
            alt="Donate Ethereum"
        />
    </a>
);

export default DonateEthereum;
