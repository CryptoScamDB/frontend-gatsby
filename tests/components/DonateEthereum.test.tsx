import React from 'react';
import renderer from 'react-test-renderer';
import DonateEthereum from '../../src/components/icons/donate/DonateEthereum';

describe('DonateEthereum', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DonateEthereum />).toJSON()!;
    const {
      props: { href }
    } = tree;
    expect(typeof href).toBe('string');
    expect(tree).toMatchSnapshot();
  });
});
