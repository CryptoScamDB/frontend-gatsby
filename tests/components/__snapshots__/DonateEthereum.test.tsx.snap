// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonateEthereum renders correctly 1`] = `
.c0 {
  height: 26px;
  width: 26px;
}

<a
  href="https://etherscan.io/address/0x7030B43E821BA34124915bA65B90Dbda7a9E7d09"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Donate Ethereum"
    className="c0"
    src="test-file-stub"
    title="cryptoscamdb.eth"
  />
</a>
`;
